### DISCUSSION TITLE
POST endpoint contract

### DISCUSSION THREAD

**Discussion by rishighan** [bpepple/metron#91] [API Suggestions]
Hello, I am at a point in my UI development, where I have the following model backing the form for editing metadata which:

1. can be written to Metron, if the user authenticated with it
2. can we written to ThreeTwo's local datastore

Note that in the model `json` below, the `value` and `label` keys are both from Metron.
Here's the model, roughly based on the same fields seen in the `Edit Issue` page in Metron:

```json
{
  "issue_name": "X-O Manowar 032 (2015) (Digital) (Mephisto-Empire)",
  "issue_number": "32",
  "issue_year": "2015-02-19T08:00:00.000Z",
  "page_count": "32",
  "description": "X-O Manowar goes to war with people that like war and dislike losing.",
  "distributor_sku": "12132123",
  "upc_code": "123123",
  "publisher": {
    "label": "AfterShock Comics",
    "value": 10
  },
  "story_arc": {
    "label": "A Court of Owls",
    "value": 177
  },
  "series": {
    "label": "1st Issue Special (1975)",
    "value": 859
  },
  "credits": [
    {
      "creator": {
        "label": "A. J. Lieberman",
        "value": 2916
      },
      "role": {
        "label": "Interviewer",
        "value": 27
      }
    },
    {
      "creator": {
        "label": "A.J. Fierro",
        "value": 4392
      },
      "role": {
        "label": "Penciller",
        "value": 3
      }
    }
  ]
}

```

Couple of questions:

1. What are the `field` names in Metron? I want to match them in my model, so there is minimal transformation required both on Metron and ThreeTwo's side
2. Am I missing fields?

Paging @btxgit for his inputs

**Comment by rishighan** [comment]
![Screen Shot 2022-01-06 at 8 46 53 PM](https://user-images.githubusercontent.com/414927/148493268-2b9b89e2-2390-4029-ae38-eca43940decb.png)

**Comment by bpepple** [comment]
>     1. What are the `field` names in Metron? I want to match them in my model, so there is minimal transformation required both on Metron and ThreeTwo's side

Don't have much time for a detailed response to this discussion right now since I've got some work stuff to do, but you can find the models that Metron uses here: https://github.com/bpepple/metron/tree/master/comicsdb/models

**Reply by rishighan** [reply]
Thank you! I'll take a look

**Comment by rishighan** [comment]
Hello @bpepple I am bumping this thread to resurrect this conversation again. 
`TLDR;` the ability for authenticated users of Metron to be able to contribute to Metron from within ThreeTwo.

Are you at a place where this discussion can continue? Even if you have alternative approaches I am willing to discuss.

**Reply by bpepple** [reply]
I've written a [command-line tool](https://codeberg.org/bpepple/barda), that has addressed most of the issues I had identified with granting POST/PATCH permissions:

- Sanitization of data for consumption by Metron
- Correct linking of resources (character, creator, etc)
- Adding alternative ID's (GCD, CV)
- Adding reprint information
- Upload Covers & Variants

Currently, there are 4 editors and me that have POST/PATCH permissions, and I'll probably look at identifying a few more as time goes on, but I really don't expect there to be more than 10 users with these permissions. Given that, I'm not sure it make much sense for you to invest dev time for such a small potential pool of users.

**Reply by rishighan** [reply]
Thanks @bpepple always a treat to see solid tooling around Metron.
